src/porepy/numerics/ad/equation_system.py

```diff
@@ -572,56 +584,28 @@ def get_variable_values(
                 :meth:`num_dofs`.
 
         Raises:
-            ValueError: If neither of ``time_step_index`` or ``iterate_index`` have been
-            assigned a non-None value.
-            ValueError: If both ``time_step_index`` and ``iterate_index`` have been
-            assigned a value.
-            KeyError: If no values are stored for the VariableType input.
             ValueError: If unknown VariableType arguments are passed.
 
         """
-        if time_step_index is None and iterate_index is None:
```

Comment:
get and set variable values uses now the respective ad utility functions (no code duplicate anymore)
Error handling of indices is implemented there.